scipy/fftpack/helper.py

```diff
@@ -189,6 +191,32 @@ def _init_nd_shape_and_axes(x, shape, axes):
     return np.asarray(s), np.asarray(a)
 
 
+def _iterable_of_int(x, name=None):
+    """Convert ``x`` to an iterable sequence of int
+
+    Parameters
+    ----------
+    x : value, or Iterable of values, convertible to int
+    name : str, optional
+        Name of the argument being converted, only used in the error message
+
+    Returns
+    -------
+    y : ``List[int]``
+    """
+    if not isinstance(x, Iterable):
```

Comment:
I think testing for an explicit number is safer here via `numbers.Number` type check  (or via something else). This would avoid the iterable and iterator nuances etc.